### PR TITLE
allow specifying regex in global config kubecontext

### PR DESCRIPTION
**Pull request by DanielSel**
Replicate the kubecontext matching behaviour from profiles to global config

### CODE REVIEW

File: pkg/skaffold/util/regex_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2019 The Skaffold Authors
```
Comment (nkubala) [inline pkg/skaffold/util/regex_test.go:2]: nit

```suggestion
Copyright 2020 The Skaffold Authors
```

File: pkg/skaffold/util/regex.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2019 The Skaffold Authors
```
Comment (nkubala) [inline pkg/skaffold/util/regex.go:2]: nit

```suggestion
Copyright 2020 The Skaffold Authors
```

**Comment by dgageot** [comment]
@DanielSel There's just a few nits and some doc to be added

**Comment by DanielSel** [comment]
> 
> 
> The only thing I would request is to add this to the documentation here, otherwise we'll forget to document it:
> https://skaffold.dev/docs/design/global-config/

done. sorry for the delay.

File: cmd/skaffold/app/cmd/config/set.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2019 The Skaffold Authors
+Copyright 2020 The Skaffold Authors
```
Comment (balopat) [inline cmd/skaffold/app/cmd/config/set.go:2]: we really shouldn't update these - please revert them, copyright is meant to show the creation time

Comment (DanielSel) [reply cmd/skaffold/app/cmd/config/set.go]: reverted on modified files, left 2020 on new files.

**Comment by tejal29** [comment]
> Please visit http://34.94.121.254:1313 to view changes to the docs.

Checked the docs.

**Pull request merged**